### PR TITLE
Mar 1293 - Buyer dashboard

### PR DESCRIPTION
**Pull request by ndavison**

### CODE REVIEW

File: apps/marketplace/components/BuyerDashboard/BuyerDashboard.scss [hunk 1]
```diff
@@ -1,58 +1,207 @@
...
+  }
+}
+
+@media (max-width: 899px) {
+  .header, table.resultListing {
+    margin-bottom: 20px;
+  }
+
+  table.resultListing thead, table.resultListing .empty, table.resultListing .colId, table.resultListing .colAuthor, .hideSmall {
+    display: none !important;
```
Comment (tobiasbrown) [inline apps/marketplace/components/BuyerDashboard/BuyerDashboard.scss]: As discussed, maybe check to see if `!important` is still required to override UI-Kit's rules .  If it is, a comment might be helpful here.

File: apps/marketplace/components/BuyerDashboard/BuyerDashboardTeamOverview.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      )
+    }
+
+    return (
+      <div className="row">
+        <div className="col-xs-12">
+          <h2>Active team members</h2>
+          <p>
+            If this list contains members who have left your organisation, please <a href="/contact-us">contact us</a>{' '}
+            to have them removed
```
Comment (tobiasbrown) [inline apps/marketplace/components/BuyerDashboard/BuyerDashboardTeamOverview.js]: Needs full stop.

File: apps/marketplace/components/BuyerDashboard/BuyerDashboard.scss [hunk 3]
```diff
@@ -1,58 +1,207 @@
...
 
-.buttonsContainer {
-  float: right;
-  text-align: right;
-  margin-top: -4.5em;
+table.resultListing tr:nth-child(2n) {
+  background-color: #fafafa;
+}
+
+table.resultListing .colId {
```
Comment (matt-sm) [inline apps/marketplace/components/BuyerDashboard/BuyerDashboard.scss]: all this intricate width/padding is avoided with a grid

File: apps/marketplace/components/BuyerDashboard/BuyerDashboard.scss [hunk 4]
```diff
@@ -1,58 +1,207 @@
...
+
+ul.menu li {
+  display: inline-block;
+  margin: 0;
+}
+
+ul.menu li a {
+  display: block;
+  font-size: 1.25rem;
+  padding: 10px;
```
Comment (matt-sm) [inline apps/marketplace/components/BuyerDashboard/BuyerDashboard.scss]: should these all be `rem` ?

File: apps/marketplace/components/BuyerDashboard/BuyerDashboardHeader.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+export class BuyerDashboardHeader extends Component {
+  constructor(props) {
+    super(props)
+    this.handleMenuClick = this.handleMenuClick.bind(this)
+  }
+
+  handleMenuClick(e) {
+    e.preventDefault()
+    const targetEl = e.currentTarget
+    this.props.history.push(targetEl.getAttribute('href'))
```
Comment (matt-sm) [inline apps/marketplace/components/BuyerDashboard/BuyerDashboardHeader.js]: shouldn't the router to do this?

File: apps/marketplace/components/BuyerDashboard/BuyerDashboard.scss (hunk 4 above)
Comment (ndavison) [reply apps/marketplace/components/BuyerDashboard/BuyerDashboard.scss]: Possibly - the nav/menu CSS was copied straight from the existing dashboard to try and get the same look and feel (as it's not a uitkit component anymore).

File: apps/marketplace/components/BuyerDashboard/BuyerDashboardHeader.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { rootPath } from 'marketplace/routes'
+import styles from './BuyerDashboard.scss'
+
+export class BuyerDashboardHeader extends Component {
```
Comment (matt-sm) [inline apps/marketplace/components/BuyerDashboard/BuyerDashboardHeader.js]: you could make this SFC - no need to connect it - pass the function and the org as props

File: apps/marketplace/components/BuyerDashboard/BuyerDashboardHeader.test.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import Enzyme, { mount } from 'enzyme'
+import Adapter from 'enzyme-adapter-react-16'
+import { BuyerDashboardHeader } from './BuyerDashboardHeader'
+
+Enzyme.configure({ adapter: new Adapter() })
+
+jest.mock('shared/Icon/_getIcons')
+
+test('Displays the organisation', () => {
```
Comment (matt-sm) [inline apps/marketplace/components/BuyerDashboard/BuyerDashboardHeader.test.js]: 👍 for doing some tests

File: apps/marketplace/components/BuyerDashboard/BuyerDashboardMyBriefs.js [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+import ClosedDate from 'shared/ClosedDate'
+import LoadingIndicatorFullPage from 'shared/LoadingIndicatorFullPage/LoadingIndicatorFullPage'
+import { loadBuyerDashboard } from 'marketplace/actions/dashboardActions'
+import { BUYER_DASHBOARD_MYBRIEFS_SUCCESS } from 'marketplace/constants/constants'
+import { statusConvert } from 'marketplace/components/helpers'
+import { rootPath } from 'marketplace/routes'
+import BuyerDashboardHelp from './BuyerDashboardHelp'
+import styles from './BuyerDashboard.scss'
+
+export class BuyerDashboardMyBriefs extends Component {
```
Comment (matt-sm) [inline apps/marketplace/components/BuyerDashboard/BuyerDashboardMyBriefs.js]: again this probably should be SFC - define functions and state at the Page level and pass as props

File: apps/marketplace/components/BuyerDashboard/BuyerDashboardMyBriefs.js [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  componentDidMount() {
+    this.props.loadData(BUYER_DASHBOARD_MYBRIEFS_SUCCESS, '/dashboard/my/briefs')
+  }
+
+  render() {
+    if (this.props.currentlySending) {
```
Comment (matt-sm) [inline apps/marketplace/components/BuyerDashboard/BuyerDashboardMyBriefs.js]: another example - do this once back at the parent

File: apps/marketplace/components/BuyerDashboard/BuyerDashboardHeader.js (hunk 5 above)
Comment (ndavison) [reply apps/marketplace/components/BuyerDashboard/BuyerDashboardHeader.js]: It looks like BrowserRouter and Link might do this - I'll look into those.

File: apps/marketplace/components/BuyerDashboard/BuyerDashboardMyBriefs.js (hunk 8 above)
Comment (matt-sm) [reply apps/marketplace/components/BuyerDashboard/BuyerDashboardMyBriefs.js]: Makes these components easier to test as well

File: apps/marketplace/components/BriefChoice/BriefChoice.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react'
+import { Radio } from '@gov.au/control-input/lib/js/react.js'
```
Comment (matt-sm) [inline apps/marketplace/components/BriefChoice/BriefChoice.js]: I wonder if this is easier in latest version

File: apps/marketplace/components/BuyerDashboard/BuyerDashboardMyBriefs.js [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+import { loadBuyerDashboard } from 'marketplace/actions/dashboardActions'
+import { BUYER_DASHBOARD_MYBRIEFS_SUCCESS } from 'marketplace/constants/constants'
+import { statusConvert } from 'marketplace/components/helpers'
+import { rootPath } from 'marketplace/routes'
+import BuyerDashboardHelp from './BuyerDashboardHelp'
+import styles from './BuyerDashboard.scss'
+
+export class BuyerDashboardMyBriefs extends Component {
+  componentDidMount() {
+    this.props.loadData(BUYER_DASHBOARD_MYBRIEFS_SUCCESS, '/dashboard/my/briefs')
```
Comment (matt-sm) [inline apps/marketplace/components/BuyerDashboard/BuyerDashboardMyBriefs.js]: not sure about the component knowing about this constant - should really create actions for each case

File: apps/marketplace/components/BriefChoice/BriefChoice.js (hunk 10 above)
Comment (ndavison) [reply apps/marketplace/components/BriefChoice/BriefChoice.js]: I don't think so - it seems the latest uikit still points its package.json `main` attribute to the es5 code.

**Pull request merged**